src/Application/Messenger/Handler/SendRequestConsumer.php

```diff
@@ -56,9 +58,24 @@ public function __invoke(SendRequest $sendRequest): void
             $this->fail($sendRequest->id, "Validation failed: {$this->validator->getReason()}");
         }
 
-        $bodyRenderedHtml = $this->twig->render("{$sendRequest->templateKey}.html.twig", $sendRequest->params);
+        // Instantiate a new Swift Message object
+        $email = new \Swift_Message();
+        $cssToInlineStyles  = new CssToInlineStyles();
+
+        $images['headerImageRef'] = $this->embedImages($email, 'TBG.jpg');
+        $images['footerImageRef'] = $this->embedImages($email, 'CCh.jpg');
+
+        $templateMergeParams = array_merge($images, $sendRequest->params);
+
+        $bodyRenderedHtml = $this->twig->render("{$sendRequest->templateKey}.html.twig", $templateMergeParams);
         $bodyPlainText = strip_tags($bodyRenderedHtml);
 
+        $cssToInlineStyles->setUseInlineStylesBlock(false)
```

Comment:
@NoelLH Can you have a look and see whether i'm doing this right? I've followed the guidance in the readme `https://github.com/OpenBuildings/swiftmailer-css-inliner` but it's doesn't seem to be working as expected, i.e no emails are being sent at all

Reply:
@tbg-jasonfung have you tried just leaving the config as it is? The plugin is already registered with SwiftMailer, and pretty sure it was working with my original example – I changed `p` size to check. No PHP changes needed.

If the email isn't sending, you'll have an error somewhere. Check the container logs to isolate the problem.

Reply:
@NoelLH It does work, however it shows the styling tags in the preview: 
![image](https://user-images.githubusercontent.com/47176130/82456859-ab775b80-9aac-11ea-9fd4-434940585751.png)

Reply:
Not really enough info here @tbg-jasonfung, I would print out the HTML on the app side and look for the error. Also check if the plain text part tag stripping is working.

Reply:
@NoelLH I've printed out the plain text part with tags removed, looks like it's kept the css classes, is there also a way to strip these out?

```
string(1434) "

      p, ul, li {
        font-size: 0.9rem;
      }

      blockquote {
        font-size: 0.7rem;
        font-style: italic;
      }

      .container {
        margin: 0 auto;
        max-width: 500px;
      }

      .content {
        text-align: left;
      }

      .headerImg {
        max-width: 50%;
        height: auto;
      }

      .footerImg {
        max-width: 30%;
        height: auto;
      }

        Dear Patti Smith,
  Thank you for supporting The charity with your generous gift through the Big Give.

    Here is your donation information:

        Total for The charity: £600.02
        Donation: £400.01
        Gift Aid: £100
                    Match Funds: £200.01

    Donation reference: d290f1ee-6c54-4b01-90e6-d701748f0851
    Reference on credit/debit card statement: CharityCheckout.co.uk

      Thank you for leaving a tip of £10 toward the Big Give.

  Total amount you have been charged is £410.01.

      A message from The charity:
    This is a really long message from the charity This is a really long message from the charity This is a really long message from the charity

  If you have any questions, please contact us.

          Kind regards,
          The Big Give Team
          www.thebiggive.org.uk

        Powered by Charity Checkout
```

Reply:
Ah yes, makes sense @tbg-jasonfung if you read the docs https://www.php.net/manual/en/function.strip-tags.php I misremembered the scope of that fn.

Lots of ways around this. A regex to remove everything outside the content div after rendering might be quickest, though not very elegant or easy to maintain.

We could also think about conditionally extending a different base depending on the context, and render twice. We're going to discard everything outside the inner template except maybe the footer sign-off copy, so it's probably more maintainable to have a different base template which makes that explicit and lets us manage those parts independently if desired. We could pass in a `renderFormat` context var and conditionally extend either a `_base.html.twig` or `_base.text.twig`. Notes on that [here](https://ourcodeworld.com/articles/read/467/how-to-conditionally-extend-a-template-or-import-a-macro-in-twig).

Reply:
@NoelLH I've literally just added a preg_replace + expression, I think this should achieve without the additional base template, let me know if you think otherwise

Reply:
@tbg-jasonfung I think a distinct base is more maintainable as explained above.